modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryPerformanceTest.java

```diff
@@ -132,33 +133,185 @@ private void _addObjectEntries(Integer numberOfEntries) throws Exception {
+				18000,
+				StringBundler.concat(
+					" Import ", _objectEntryCount, " Object Entries"))) {
+
+			_invokeHttp(
+				_createObjectEntryJSON(), HttpInvoker.HttpMethod.POST,
+				_getPath(_PATH_SUFFIX));
+
+			List<ObjectDefinition> objectDefinitions =
+				_objectDefinitionLocalService.getCustomObjectDefinitions(
+					_CUSTOM_OBJECT_DEFINITION_STATUS);
+
+			_customObjectDefinition = objectDefinitions.get(
+				_OBJECT_DEFINITION_LIST_INDEX);
+
+			long currentObjectEntryCount = 0;
+
+			while (currentObjectEntryCount < _objectEntryCount) {
+				currentObjectEntryCount =
+					_objectEntryLocalService.getObjectEntriesCount(
+						_customObjectDefinition.getObjectDefinitionId());
+			}
+		}
+
+		try (PerformanceTimer performanceTimer = new PerformanceTimer(
+				15000,
+				StringBundler.concat(
+					" Delete ", _objectEntryCount, " Object Entries"))) {
+
+			HttpInvoker.HttpResponse httpResponse = _invokeHttp(
+				null, HttpInvoker.HttpMethod.GET,
+				_getPath("/o/c/foos/?fields=id&pageSize=" + _objectEntryCount));
+
+			JSONObject jsonObject = _jsonFactory.createJSONObject(
+				httpResponse.getContent());
+
+			JSONArray jsonArray = (JSONArray)jsonObject.get("items");
+
+			_invokeHttp(
+				jsonArray.toString(), HttpInvoker.HttpMethod.DELETE,
+				_getPath(_PATH_SUFFIX));
+
+			long currentObjectEntryCount = _objectEntryCount;
+
+			while (currentObjectEntryCount != 0) {
+				currentObjectEntryCount =
+					_objectEntryLocalService.getObjectEntriesCount(
+						_customObjectDefinition.getObjectDefinitionId());
+			}
+		}
 	}
 
-	private void _deleteObjectEntries() throws Exception {
-		for (com.liferay.object.model.ObjectEntry objectEntry :
-				_objectEntries) {
+	private String _createObjectEntryJSON() {
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray();
 
-			_objectEntryLocalService.deleteObjectEntry(objectEntry);
+		for (int i = 0; i < _objectEntryCount; i++) {
+			jsonArray.put(JSONUtil.put("alpha", "foo"));
 		}
+
+		return jsonArray.toString();
 	}
 
-	private static Path _logFilePath;
+	private String _getPath(String pathSuffix) {
+		return StringBundler.concat(
+			"http://", _VIRTUAL_HOST_NAME, ":8080", pathSuffix);
+	}
+
+	private HttpInvoker.HttpResponse _invokeHttp(
+			String bodyJSON, HttpInvoker.HttpMethod httpMethod, String path)
+		throws Exception {
+
+		HttpInvoker httpInvoker = HttpInvoker.newHttpInvoker();
+
+		if (bodyJSON != null) {
+			httpInvoker.body(bodyJSON, "application/json");
+		}
+
+		httpInvoker.httpMethod(httpMethod);
+		httpInvoker.path(path);
+		httpInvoker.userNameAndPassword(
+			StringBundler.concat(
+				"test@", _VIRTUAL_HOST_NAME, ":",
+				PropsValues.DEFAULT_ADMIN_PASSWORD));
+
+		return httpInvoker.invoke();
+	}
+
+	private static final int _CUSTOM_OBJECT_DEFINITION_STATUS = 0;
+
+	private static final int _OBJECT_DEFINITION_LIST_INDEX = 0;
+
+	private static final String _PATH_SUFFIX = "/o/c/foos/batch";
+
+	private static final String _VIRTUAL_HOST_NAME = "www.able.com";
```

Comment:
@slnn @dantewang Why we need hardcode this to "www.able.com", this is so confusing, that's why Victor's local test did not pass, it is hard to remember you have to set this to your hosts.